src/base/bittorrent/session.cpp

```diff
@@ -2431,9 +2451,21 @@ const QStringList Session::getListeningIPs()
     }
 
     if (ifaceName.isEmpty()) {
-        if (!ifaceAddr.isEmpty())
-            IPs.append(ifaceAddr);
-        else
+        if (!ifaceAddr.isEmpty()) {
+            if (!ifaceAddrPattern.isEmpty() && !addressPatternMatches(ifaceAddr, ifaceAddrPattern)) {
+                logger->addMessage(tr("Configured network interface address %1 doesn't match the given pattern.", "Configured network interface address 124.51.168.1 doesn't match the given pattern.").arg(ifaceAddr), Log::CRITICAL);
+                IPs.append("127.0.0.1"); // Force listening to localhost and avoid accidental connection that will expose user data.
+                return IPs;
+            }
+            else {
+                IPs.append(ifaceAddr);
+            }
+        }
+        else if (!ifaceAddrPattern.isEmpty()) {
+            logger->addMessage(tr("No network interface/address is specified, but address pattern is set."), Log::CRITICAL);
+            IPs.append("127.0.0.1"); // Force listening to localhost and avoid accidental connection that will expose user data.
+        }
+        else // no constraints, listen on any interface/address
```

Comment:
This `else` should actually have curly braces since the conditionals before it make use of them.